packages/Frontend/src/containers/AppContainer.jsx

```diff
@@ -29,14 +21,28 @@ class AppContainer extends Component {
     };
   }
 
+  componentDidMount() {
```

Comment:
When the AppContainer is put on the screen (i.e. mounted) this bit of code runs. I found out how to do it in a part of this guide: https://firebase.google.com/docs/auth/web/manage-users#get_the_currently_signed_in_user. I read about it a little more in detail in the reference: https://firebase.google.com/docs/reference/js/firebase.auth.Auth#onAuthStateChanged

To put it simply, any time anything changes anywhere in the app with authentication this function runs. So when we click the button in the sidebar that logs us out, this function runs. If anything changes, but the user is still signed in or is signing in then, then user in the function is that user. If the user logs out, the user argument is null.

That makes it so I can say, if there's a user, use this.props.login, which puts them in redux (it should probably be renamed from login, to setUser or something clearer), otherwise, they're logging out, call this.props.logout() which is a redux function to remove the user from redux.